# Experiment 01: Implement a banana camel problem 

## Aim:
A person has 3000 bananas and a camel. The person wants to transport the maximum number of
bananas to a destination which is 1000 KMs away, using only the camel as a mode of transportation.
• The camel cannot carry more than 1000 bananas at a time and eats a banana every km it travels.
• What is the maximum number of bananas that can be transferred to the destination using only
camel (no other mode of transportation are allowed)

## Program

In [1]:
total_bananas = int(input('No. of bananas at start: '))
distance = int(input('Distance to be covered: '))
load_capacity = int(input('Maximum bananas camel can carry at at a time:'))
bananas_lost = 0
start = total_bananas
for i in range(distance):
    while start>0:
        start = start-load_capacity
        if start == 1:
            bananas_lost = bananas_lost - 1
        bananas_lost = bananas_lost + 2
    bananas_lost = bananas_lost - 1
    start = total_bananas - bananas_lost
    if start == 0:
        break
print("Total bananas delivered:",start)

No. of bananas at start: 3000
Distance to be covered: 1000
Maximum bananas camel can carry at at a time:1000
Total bananas delivered: 533


## Manual Output: Manual calculation for the example you have taken:
Since there are 3000 bananas and the Camel can only carry 1000 bananas, he will have to make 3 trips
to carry them all to any point in between.
<---p1---><--------p2-----><-----p3---->
A---------------------------------------->B
When bananas are reduced to 2000 then the Camel can shift them to another point in 2 trips and
when the number of bananas left are <= 1000, then he should not return and only move forward.
In the first part, P1, to shift the bananas by 1Km, the Camel will have to
Move forward with 1000 bananas – Will eat up 1 banana in the way forward
Leave 998 banana after 1 km and return with 1 banana – will eat up 1 banana in the way back
Pick up the next 1000 bananas and move forward – Will eat up 1 banana in the way forward
Leave 998 banana after 1 km and return with 1 banana – will eat up 1 banana in the way back
Will carry the last 1000 bananas from point a and move forward – will eat up 1 banana
Note: After point 5 the Camel does not need to return to point A again.
So to shift 3000 bananas by 1km, the Camel will eat up 5 bananas.
After moving to 200 km the Camel would have eaten up 1000 bananas and is now left with 2000
bananas.
Now in the Part P2, the Camel needs to do the following to shift the Bananas by 1km.
Move forward with 1000 bananas – Will eat up 1 banana in the way forward
Leave 998 banana after 1 km and return with 1 banana – will eat up this 1 banana in the way back
Pick up the next 1000 bananas and move forward – Will eat up 1 banana in the way forward
Note: After point 3 the Camel does not need to return to the starting point of P2.
So to shift 2000 bananas by 1km, the Camel will eat up 3 bananas.
After moving to 333 km the camel would have eaten up 1000 bananas and is now left with the last
1000 bananas.
The Camel will actually be able to cover 333.33 km, I have ignored the decimal part because it will not
make a difference in this example.
Hence the length of part P2 is 333 Km.
Now, for the last part, P3, the Camel only has to move forward. He has already covered 533 (200+333)
out of 1000 km in Parts P1 & P2. Now he has to cover only 467 km and he has 1000 bananas.
He will eat up 467 bananas on the way forward, and at point B the Camel will be left with only 533
Bananas.